bip-0000.mediawiki

```diff
@@ -0,0 +1,335 @@
+
+After the inputs have been selected, the sender can create one or more outputs for one or more silent payment addresses in the following manner:
+
+* Let ''a = a<sub>0</sub> + a<sub>1</sub> + … a<sub>n</sub>'', where each ''a<sub>i</sub>'' is the 256-bit private key(s) of an input from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+* Group receiver silent payment addresses by ''B<sub>scan</sub>'' (e.g. each group consists of one ''B<sub>scan</sub>'' and one or more ''B<sub>m</sub>'')
+* For each group:
+** Let ''ecdh_shared_secret = outpoints_hash·a·B<sub>scan</sub>''
+** Let ''n = 0''
+** For each ''B<sub>m</sub>'' in the group:
+*** Let ''t<sub>n</sub> = sha256(ecdh_shared_secret || ser<sub>32</sub>(n))''
+*** Let ''P<sub>mn</sub> = t<sub>n</sub>·G + B<sub>m</sub>''
+*** Encode ''P<sub>mn</sub>'' as a [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP341] taproot output
+*** Optionally, repeat with n++ to create additional outputs for the current ''B<sub>m</sub>''
+*** If no additional outputs are required, continue to the next ''B<sub>m</sub>'' with ''n++''
+
+The sender should check to ensure that no two outputs in the final transaction have the same ''t<sub>n</sub>'' as this could result in a loss of privacy for the receiver<ref name="why_not_the_same_tn">''' Why can't two outputs have the same ''t<sub>n</sub>''?''' If paying the same entity but to two separate labeled addresses in the same transaction without incrementing ''n'', the two outputs would be ''t<sub>n</sub>·G + B<sub>spend</sub> + i·G'' and ''t<sub>n</sub>·G + B<sub>spend</sub> + j·G''. The attacker could subtract the two values and observe that the distance between i and j is small. This would allow them to deduce that this transaction is a silent payment transaction and that a single entity received two outputs, but won't tell them who the entity is.</ref>.
+
+=== Receiver ===
+
+==== Key Derivation ====
+
+Two keys are needed to create a silent payments address: the spend key and the scan key. While these keys can be generated independently, wallet software SHOULD use BIP32 derivation<ref name="bip32_derivation">'''Why use BIP32 hardened derivation?''' Using BIP32 derivation allows users to add silent payments to an existing master seed. It also ensures that a user's silent payment funds are recoverable in any wallet which supports BIP32 derivation. Using hardened derivation ensures that it is safe to export the scan private key without exposing the master key or spend private key.</ref> to ensure compatibility across wallets.
+
+A scan and spend key pair using BIP32 derivation are defined (taking inspiration from [https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki BIP44]) in the following manner:
+
+
+    Spend private key: m / purpose' / coin_type' / account' / silent_payment_version' / 0' / 0
+     Scan private key: m / purpose' / coin_type' / account' / silent_payment_version' / 1' / 0
```

Comment:
Why does the silent payment version need to be in the derivation path? It seems to me that a future version would be its own BIP and therefore purpose field.

Reply:
This is here precisely because I _don't_ see us needing a new BIP for new versions of silent payments. Imagine we get a new output type and want to add it to the "Inputs for shared secret derivation" list. All we'd need to do is update this BIP with a version 1 section and everything else would remain unchanged.

However, the more I think about this, maybe having the version in the derivation path is a bad idea. Imagine the same scenario I described above: for this upgrade, it would be great if you could keep the same secret data and backups, and just update your address with a new version number (`sp1p...`). This sp v1 address could still accept payments from v0 wallets, as well as v1 wallets.

If there is a different upgrade to silent payments which fundamentally changes how the address is used (e.g using something besides x-only pubkeys), then we would likely have a new BIP number, which makes the version field unnecessary.

Thoughts @RubenSomsen ?

Reply:
The more I've thought about this, I think we should remove the version from the derivation path since having it would require users to generate new keys when there is likely no reason to.

 I've updated the BIP to reflect this and also added a blurb to the version section.